src/weights.jl

```diff
@@ -21,6 +21,7 @@ length(wv::AbstractWeights) = length(wv.values)
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 size(wv::AbstractWeights) = size(wv.values)
+Base.axes(wv::AbstractWeights) = Base.axes(wv.values)
```

Comment:
Is this definition correct? We only support `Integer` indices in `getindex` and `Int` in `setindex!` below it seems, so non-standard axes that don't operate with `Integer`s will yield unsupported indices it seems.

Reply:
The `AbstractArray` interface requires that `axes` returns a tuple of `AbstractUnitRange{<:Integer}`s, so we can assume that their `eltype` will be an `Integer`. From the `axes` docstring: "Each of the indices has to be an `AbstractUnitRange{<:Integer}`".

As for `Integers` other than `Int`, As long as we implement `getindex` and `setindex` for `Int` as required by the AbstractArray interface, Julia's Base libraries will convert other `Integer`s to `Int` as needed.

For example
```julia
using StatsBase
w = Weights([4,2,6])
w[0x2] = 1
display(w)
#=
3-element Weights{Int64, Int64, Vector{Int64}}:
 4
 1
 6
=#
```

The previous implicit definition (`axes(wv::AbstractWeights) = (Base.OneTo(length(wv)),)`) was _not_ correct

```julia
using StatsBase, OffsetArrays
w = Weights(OffsetVector([4,2,6], 2))
display(w)
#=
3-element Weights{Int64, Int64, OffsetVector{Int64, Vector{Int64}}}:
          3 # unsafe read
 4972085568 # unsafe read
          4
=#
```
On this PR
```julia
using StatsBase, OffsetArrays
w = Weights(OffsetVector([4,2,6], 2))
display(w)
#=
3-element Weights{Int64, Int64, OffsetVector{Int64, Vector{Int64}}} with indices 3:5:
 4
 2
 6
=#
```

Reply:
Given the discussion in #791 I would propose to add tests of the consequences of this change (e.g. making sure that `getindex`, setindex!`, `firstindex`, `lastindex`, `isequal` etc. work correctly for non-1-based case)

Reply:
if StatsBase.jl maintainers are OK to merge this PR without these tests then I can add them later.